pkg/cmd/step/boot/step_boot_vault.go

```diff
@@ -132,8 +136,43 @@ func (o *StepBootVaultOptions) Run() error {
 		KeyName:             requirements.Vault.Key,
 		ServiceAccountName:  requirements.Vault.ServiceAccount,
 		ClusterName:         requirements.Cluster.ClusterName,
-		// TODO - load from a local yaml file if available?
-		// AWSConfig:           o.AWSConfig,
+	}
+	if requirements.Cluster.Provider == cloud.EKS {
+		awsConfig := requirements.Vault.AWSConfig
+		cvo.AWSCreateVaultOptions = create.AWSCreateVaultOptions{}
+		if !awsConfig.AutoCreate && (checkRequiredResource("dynamoDBTable", awsConfig.DynamoDBTable) ||
+			checkRequiredResource("providedIAMUsername", awsConfig.ProvidedIAMUsername) ||
+			checkRequiredResource("secretAccessKey", awsConfig.SecretAccessKey) ||
+			checkRequiredResource("accessKeyId", awsConfig.AccessKeyID) ||
+			checkRequiredResource("kmsKeyId", awsConfig.KMSKeyID) ||
+			checkRequiredResource("s3Bucket", awsConfig.S3Bucket)) {
+			log.Logger().Info("Some of the required provided values are empty - We will create all resources")
+			awsConfig.AutoCreate = true
+		}
+		cvo.Boot = true
+		cvo.AWSConfig = kubevault.AWSConfig{
+			AWSUnsealConfig: v1alpha1.AWSUnsealConfig{
+				KMSKeyID:  awsConfig.KMSKeyID,
+				KMSRegion: awsConfig.KMSRegion,
+				S3Bucket:  awsConfig.S3Bucket,
+				S3Prefix:  awsConfig.S3Prefix,
+				S3Region:  awsConfig.S3Region,
+			},
+			AutoCreate:          awsConfig.AutoCreate,
+			DynamoDBTable:       awsConfig.DynamoDBTable,
+			DynamoDBRegion:      awsConfig.DynamoDBRegion,
+			AccessKeyID:         awsConfig.AccessKeyID,
+			SecretAccessKey:     awsConfig.SecretAccessKey,
+			ProvidedIAMUsername: awsConfig.ProvidedIAMUsername,
+		}
+		cvo.AWSTemplatesDir = filepath.Join(o.Dir, o.ProviderValuesDir, cloud.EKS, "templates")
+		defer func() {
+			// we don't want to save secret keys so we have to wipe them before saving the file
+			log.Logger().Info("Deleting AccessKeyID and SecretAccessKey from the jx-requirements.yml file")
+			requirements.Vault.AWSConfig.AccessKeyID = ""
```

Comment:
This is a bit dangerous. Are the requirements saved on disk with the access key until this function gets executed to clear them up?

You can also ignore the fields from serialisation using the tag:
```
json:"-"`
```

Reply:
They are never saved on disk at this point, that's why I want to run this and save the file after they've been used.
This is just a way to let users provide the access keys instead of an user.

Reply:
I think this might be dangerous because if the jx-requirements file gets somehow saved in git, these credentials might get leaked. 

Why not to use the env variables for them? They don't really belong to the requirements since they should never be saved, and are somehow ephemeral.  

I would strongly suggest to remove them and to use env variables instead, and maybe command line arguments but I think the standard way is to provide them in env variables.